### ISSUE TITLE
LISTEN / NOTIFY

### ISSUE THREAD

**Issue by vitaly-t** [jameshy/pg-job-queue#19]
Interestingly, I just came across an older project like this one: [node-pg-jobs](https://github.com/noblesamurai/node-pg-jobs), and was wondering whether you saw it before, and what are the differences.

It has an open issue [LISTEN/NOTIFY for better performance](https://github.com/noblesamurai/node-pg-jobs/issues/20), which I quite liked.

And in case you decide to incorporate something like that into this library, there are now good instructions for this within `pg-promise`: [LISTEN / NOTIFY](https://github.com/vitaly-t/pg-promise/wiki/Learn-by-Example#listen--notify).

**Comment by jameshy** [comment]
Yep i've seen [node-pg-jobs](https://github.com/noblesamurai/node-pg-jobs).

The main differences so far, to the best of my knowledge:

**This project uses promises rather than callbacks**
Supported by the excellent [pg-promise](https://github.com/vitaly-t/pg-promise) library!
This leads to cleaner code.  If the 'job handler' throws an exception, or returns a failing Promise, we handle that.

**All jobs must have a 'type' in this project**
With [node-pg-jobs](https://github.com/noblesamurai/node-pg-jobs) there is no such restriction.
Having a 'type' seems nicer to me, because it helps organise job handler code (at least in my project).
Also, you could run multiple job processors, each one targeting different job types (e.g. a job processor for 'sendEmail' jobs, and another for 'processData').  You could run these on different machines.

**This project has the tool `process-jobs` for easily launching a robust job-processor**
You point the script at a handlers.js definition file, and it runs forever.
You can optionally configure special handlers in the handlers.js file, for example if you wanted to log all errors to a centralised logging solution.

General goals are:
- Don't cause drastic performance problems to postgresql.
- Really quick and easy to get up and running.
- Stable and robust, with nice handling for failed jobs (e.g. a remote service was offline, we should try again in 3 minutes)
- It should be possible to reschedule jobs (e.g. process some data from an FTP server, every 5 minutes)

Regarding LISTEN/NOTIFY.  I would love that  feature!
Currently It's not a high priority for me (it works fine already).  But definitely I will implement that in the future, if nobody else does ;)

**Comment by vitaly-t** [comment]
Nice! :+1:
